packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -689,3 +690,68 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
     );
   }
 }
+
+/// A convenience class that wraps a [DropdownButton] in a [FormField] so as to
+/// act as a part of a [Form].
+///
+/// The [DropdownButton] widget is displayed without the underline, because it
+/// is provided by the [InputDecorator].
+///
+/// See also:
+///  * [DropdownButton], which is the underlying Widget providing the functionality
+///  * [InputDecorator], which shows the visual elements and styles the form
+///    field
+class DropdownButtonFormField<T> extends FormField<T> {
+  /// Creates a [DropdownButton] widget wrapped in an [InputDecorator] and
+  /// [FormField].
+  ///
+  /// As per [DropdownButton] items and onChanged must be supplied.
+  DropdownButtonFormField({
+    Key key,
+    T value,
+    @required List<DropdownMenuItem<T>> items,
+    @required this.onChanged,
+    InputDecoration decoration,
+    FormFieldSetter<T> onSaved,
+    FormFieldValidator<T> validator,
+    Widget hint,
+  }) : super(
+    key: key,
+    onSaved: onSaved,
+    validator: validator,
+    builder: (FormFieldState<T> field) {
+      final InputDecoration effectiveDecoration = (decoration ?? const InputDecoration())
+        .applyDefaults(Theme.of(field.context).inputDecorationTheme);
+      return new InputDecorator(
+        decoration: effectiveDecoration.copyWith(errorText: field.errorText),
+        child: new DropdownButtonHideUnderline(
+          child: new DropdownButton<T>(
+              isDense: true,
+              value: value,
+              items: items,
+              hint: hint,
+              onChanged: field.didChange,
+          ),
+        ),
```

Comment:
Need to add `isEmpty: value == null` to the InputDecorator otherwise the label shows as though a value is set when one isn't.